mypyc/irbuild/classdef.py

```diff
@@ -565,7 +566,17 @@ def add_non_ext_class_attr_ann(
     if typ is None:
         # FIXME: if get_type_info is not provided, don't fall back to stmt.type?
         ann_type = get_proper_type(stmt.type)
-        if isinstance(ann_type, Instance):
+        if (
+            isinstance(stmt.unanalyzed_type, UnboundType)
+            and stmt.unanalyzed_type.original_str_expr is not None
+        ):
+            # Annotation is a forward reference, so don't attempt to load the actual
+            # type and load the string instead.
+            #
+            # TODO: is it possible to determine whether a non-string annotation is
+            # actually a forward reference due to the __annotations__ future?
```

Comment:
If `from __future__ import annotations` is used, I think that we can always use string annotations, or whenever there is any doubt that type might contain forward references.

Reply:
Ah yeah you're right. I'll open an issue about hardening our `__annotations__` handling overall when I have time.